src/patterns/Pagination/Pagination.js

```diff
@@ -83,6 +86,45 @@ export default {
       this.$emit('onPageChange', { newStart: this.newStart });
     },
 
+    pageSelector() {
+      /**
+       * When user clicks on the page selector
+       *
+       * @event pageSelector
+       * @type none
+       */
+      this.$emit('pageSelector');
+    },
+
+    pageSelected(page) {
+      /**
+       * When user selects a page from the popup menu
+       *
+       * @event pageSelected
+       * @type none
+       */
+      this.emit('pageSelected', { page });
+    },
+
+    nextPage() {
+      /**
+       * When user clicks on the next page button
+       *
+       * @event nextPage
+       * @type none
+       */
+      this.$emit('nextPage');
+    },
+
+    previousPage() {
+      /**
+       * When user clicks on the prevous page button
+       *
+       * @event previousPage
+       * @type none
+       */
+      this.$emit('previousPage');
```

Comment:
Couldn't you cut all four of these methods and simply emit from the HTML parameter?